src/generator.cs

```diff
@@ -3908,6 +3910,12 @@ public void GenerateMethodBody (MemberInformation minfo, MethodInfo mi, string s
 
 		CurrentMethod = String.Format ("{0}.{1}", type.Name, mi.Name);
 
+		// Warn about [Static] used in a member of [Category]
+		var baseTypeAtt = AttributeManager.GetCustomAttribute<BaseTypeAttribute> (minfo.type);
```

Comment:
You only need `baseTypeAtt` to show the error, so you can remove this line inside the `if` statement so it's not always called.